libs/ledger/src/chain/main_chain.cpp

```diff
@@ -388,7 +388,7 @@ void MainChain::KeepBlock(IntBlockPtr const &block) const
     auto forward_refs{forward_references_.equal_range(hash)};
     for (auto ref_it{forward_refs.first}; ref_it != forward_refs.second; ++ref_it)
     {
-      auto const &child{ref_it->second};
+      auto const child{ref_it->second};
```

Comment:
This actually performs a hash copy on each iteration. An alternative would be to only perfrom copy before the call to `CacheReference` below, or even smth like change the call to `CacheReference` to `CacheReferecne(hash, record.next_hash, true);`.

Reply:
This seems like the cleanest (minimal) solution to me - as I say I think there should be wider refactors (the forward reference stuff should probably be managed in its own class). I don't think the copy should matter here since byte array is shared and so there isn't really an underlying memory copy.